packages/back-end/src/services/datasource.ts

```diff
@@ -88,3 +88,44 @@ export async function testDataSourceConnection(
   const integration = getSourceIntegrationObject(datasource);
   await integration.testConnection();
 }
+
+export async function testQuery(
+  datasource: DataSourceInterface,
+  sql: string,
+  minExperimentLength: number,
+  startDate?: Date,
+  endDate?: Date,
+  experimentId?: string
+  // eslint-disable-next-line
+): Promise<any> {
+  const integration = getSourceIntegrationObject(datasource);
+  return integration.testQuery
+    ? await integration.testQuery(
+        sql,
+        minExperimentLength,
+        startDate,
+        endDate,
+        experimentId
+      )
+    : null;
+}
+
+export function getSelectQueryColumns(sql: string): string[] {
```

Comment:
Should we test this?
Also need to take another look and see if there are better ways to handle this